src/libraries/Common/src/Interop/Unix/System.Native/Interop.MemfdCreate.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_MemfdCreate", StringMarshalling = StringMarshalling.Utf8, SetLastError = true)]
+        internal static partial SafeFileHandle MemfdCreate(string name);
+
+        [LibraryImport(Libraries.SystemNative, EntryPoint = "SystemNative_MemfdSupported", SetLastError = true)]
+        private static partial int MemfdSupportedImpl();
+
+        private static volatile int s_memfdSupported = -1;
+
+        internal static bool MemfdSupported
+        {
+            get
+            {
+                int result = Interlocked.CompareExchange(ref s_memfdSupported, -1, -1);
+                if (result == -1)
+                {
+                    result = MemfdSupportedImpl();
+                    Interlocked.Exchange(ref s_memfdSupported, result);
+                }
+                return result == 1;
```

Comment:
```suggestion
        private static volatile sbyte s_memfdSupported;

        internal static bool MemfdSupported
        {
            get
            {
                sbyte memfdSupported= s_memfdSupported;
                if (memfdSupported == 0)
                {
                    Interlocked.Exchange(ref s_memfdSupported, MemfdSupportedImpl() ? 1 : -1, 0);
                    memfdSupported = s_memfdSupported;
                }
                return memfdSupported > 0;
```
- We often use 0 as uninitialized state for statics like this to avoid the cctor
- The flag does not need to be full int, it can be just sbyte
- Interlocked.CompareExchange once the flag was initialized is unnecessary overhead.
- Use Interlocked.CompareExchange to initialize to avoid the risk of multiple threads trying to initialize it to a different value in case of failure (MemfdSupportedImpl can return false in corner cases  failure modes even when memfd is actually supported)

Reply:
My suggestion above has additional tweaks compare to the one that was applied. Do you plan to incorporate it separately?

Reply:
You mentioned above Interlocked.CompareExchange is unnecessary overhead to read the value, why is it needed if it's unnecessary?

Reply:
Its use in line 36 is unnecessary.

Reply:
It is unnecessary overhead to read the value.

It is a good idea to use it to set the value to avoid the risk of multiple threads trying to initialize it to a different value in case of intermittent failure.